### PR TITLE
Networks plugin with one route and test

### PR DESCRIPTION
**Pull request by tawalton**
This is the beginning of the networks plugin. It has one route to get the list of networks, which addresses issue, #485

### CODE REVIEW

File: mimic/rest/networks_api.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        """
+        Get an :obj:`twisted.web.iweb.IResource` for the given URI prefix;
+        implement :obj:`IAPIMock`.
+        """
+        return NetworksMock(self, uri_prefix, session_store, region).app.resource()
+
+
+class NetworksMock(object):
+    """
+    Networks Mock
```
Comment (glyph) [inline mimic/rest/networks_api.py]: We should really endeavor to write better docstrings than this :)

File: mimic/rest/networks_api.py [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+from zope.interface import implementer
+from twisted.web.server import Request
+from twisted.plugin import IPlugin
+from mimic.rest.mimicapp import MimicApp
+from mimic.catalog import Entry
+from mimic.catalog import Endpoint
+from mimic.imimic import IAPIMock
+
+
+Request.defaultContentType = 'application/json'
```
Comment (glyph) [inline mimic/rest/networks_api.py]: This is unnecessary; please remove.

File: mimic/test/test_networks.py [hunk 3]
```diff
@@ -0,0 +1,37 @@
+"""
+Tests for dns api
```
Comment (glyph) [inline mimic/test/test_networks.py]: networks API, not DNS API

File: mimic/test/test_networks.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+class NetworksTests(SynchronousTestCase):
+    """
+    Tests for networks using the Networks Api plugin.
+    """
+    def setUp(self):
+        """
+        Create a :obj:`MimicCore` with :obj: `NetworksApi` as the only plugin.
+        """
+        dns_api = NetworksApi(['DFW'])
```
Comment (glyph) [inline mimic/test/test_networks.py]: networks not DNS

File: mimic/test/test_networks.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        self.helper = APIMockHelper(self, [dns_api])
```
Comment (glyph) [inline mimic/test/test_networks.py]: networks not DNS

File: mimic/test/test_networks.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        self.root = self.helper.root
+        self.uri = self.helper.uri
+
+    def test_get_networks(self):
+        """
+
+        https://developer.rackspace.com/docs/cloud-dns/v1/developer-guide/#list-ptr-records
```
Comment (glyph) [inline mimic/test/test_networks.py]: networks not DNS

**Comment by glyph** [comment]
Looks like there were some copy-paste errors from the DNS plugin which this appears to have been cribbed from - please fix and include better references to cloud networks (and openstack if anything is relevant?)

**Comment by ksheedlo** [comment]
Should we call it `neutron_{api,plugin}` to go with the other Openstack-y mocks? We already have Nova, Glance and Heat.

File: mimic/rest/networks_api.py [hunk 7]
```diff
@@ -0,0 +1,71 @@
+"""
+Networks API Plugin
+"""
```
Comment (ksheedlo) [inline mimic/rest/networks_api.py]: Needs standard future imports now:

```python
from __future__ import absolute_import, division, unicode_literals
```

File: mimic/test/test_networks.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+        networks_api = NetworksApi(['DFW'])
+        self.helper = APIMockHelper(self, [networks_api])
+        self.root = self.helper.root
+        self.uri = self.helper.uri
+
+    def test_get_networks(self):
+        """
+        Requesting a list of networks for a tenant that does not have any will return
+        a 200 with an empty network list
+        https://developer.rackspace.com/docs/cloud-dns/v1/developer-guide/#list-ptr-records
```
Comment (ksheedlo) [inline mimic/test/test_networks.py]: This is referencing Cloud DNS, but a better link to Networks is below it. We should delete this link.

**Comment by tawalton** [comment]
**Should we call it neutron_{api,plugin} to go with the other Openstack-y mocks? We already have Nova, Glance and Heat.**

@ksheedlo According to the the open stack docs, http://developer.openstack.org/api-ref-networking-v2.html, it is called networking.

**Comment by glyph** [comment]
@tawalton - that's true but according to the openstack docs http://developer.openstack.org/api-ref-compute-v2.1.html "nova" is also called "compute".

There are two APIs: "nova networking" (which I believe is old-school) and "neutron" which is new-school.

According to the "project navigator" - http://www.openstack.org/software/project-navigator - it's still called "neutron".  So for consistency we should definitely call it "neutron".

**Comment by tawalton** [comment]
thanks @glyph  this is the first time I have seen that page. +1 to renaming it.

File: mimic/test/test_neutron.py [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+from mimic.test.fixtures import APIMockHelper
+
+
+class NeutronTests(SynchronousTestCase):
+    """
+    Tests for networks using the Neutron Api plugin.
+    """
+    def setUp(self):
+        """
+
```
Comment (glyph) [inline mimic/test/test_neutron.py]: Please fill this out explaining what is being set up.

Comment (glyph) [inline mimic/test/test_neutron.py]: On second thought this is a pretty minor nitpick, so just do it in a follow-up PR…

**Comment by glyph** [comment]
LGTM thanks for the fixes.  Please do add that `setUp` docstring in a followup.

**Pull request merged**